### DISCUSSION TITLE
Purchasing/import vs selling/export price settings

### DISCUSSION THREAD

**Discussion by tfriberg** [UtilitechAS/amsreader-firmware#736] [Ideas]
It might be so that it is how it is working today, albeit the headlines are a bit unclear.

I am purchasing energy at a fixed price but selling at spot prices.

I have set the Price region to SE3 and Currency to SEK. I interpret this as my selling price.
I have also set the fixed price to 0,675 (VAT included) and the Multiplier to 1. I am interpreting this as my purchasing price.

I believe it would be good to have a way of explicitly differentiating between the purchasing and the selling price.

The price region can't be different (and therefor need only to be set once) but the purchasing and the selling price can be any combination of fixed and spot prices.

Something along the lines of this sketch:
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/867763/8b77a331-e01e-47ee-94b6-9ae13514e1e8)

**Comment by ArnieO** [comment]
Work in progress!
Please take a look at this test version of firmware v2.3, expected to be released quite soon: https://github.com/UtilitechAS/amsreader-firmware/issues/252#issuecomment-1945517908

**Reply by tfriberg** [reply]
So looking forward to the release 😄 

I saw that I had "accidentally" included the monthly fixed net fee prices as well, that are intended to give an expected total energy bill. At least in Sweden, these monthly fixed prices are a substantial part of the final energy bill.

But any scenario (I can think of) are covered in the upcoming 2.3 release with the settings being provided. Apart from the monthly fixed fees (not related to per kWh).

Idea for fee toggle on overview.
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/867763/f4e6b286-44f3-41e7-89a4-09999ad55898)

**Comment by tfriberg** [comment]
I believe the upcoming release will cover 99.6% of what I want. I can add the monthly fixed fees in Home Assistant. Great job!

**Reply by ArnieO** [reply]
This subject is indeed quite complex, as we all want a model that is BOTH easy to use AND can cover any variant of costs and fees (and there are MANY!) - which is quite a task...

**Reply by tfriberg** [reply]
As an experienced UX designer and information architect, you get my **finest stamp of approval** 🥇🥇🥇🥇🥇 of what you have accomplished. 

Particularly given the complexity with each country's different application of tariffs, taxes, and fees.

**Reply by ArnieO** [reply]
> As an experienced UX designer and information architect, you get my **finest stamp of approval** 🥇🥇🥇🥇🥇 of what you have accomplished.

Wow, thank you! 
And all the credit goes to @gskjold - the mastermind behind that.
